### PR TITLE
wip: Alternative Storage Target support

### PR DESCRIPTION
**Pull request by timyates**

### CODE REVIEW

File: microstream/src/main/java/io/micronaut/microstream/s3/DefaultS3StorageConfigurationProvider.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import io.micronaut.core.annotation.Nullable;
+
+/**
+ * @author Tim Yates
+ * @since 2.0.0
+ */
+@EachProperty("microstream.s3.storage")
+public class DefaultS3StorageConfigurationProvider implements S3StorageConfigurationProvider {
+
+    @Nullable
```
Comment (sdelamo) [inline microstream/src/main/java/io/micronaut/microstream/s3/DefaultS3StorageConfigurationProvider.java:30]:
```suggestion
    @NonNull
```

File: microstream/src/main/java/io/micronaut/microstream/s3/DefaultS3StorageConfigurationProvider.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+     * Class of the Root Instance.
+     * <a href="https://docs.microstream.one/manual/storage/root-instances.html">Root Instances</a>
+     * @param rootClass Class for the Root Instance.
+     */
+    public void setRootClass(@NonNull Class<?> rootClass) {
+        this.rootClass = rootClass;
+    }
+
+    @Override
+    public String getBucketName() {
```
Comment (sdelamo) [inline microstream/src/main/java/io/micronaut/microstream/s3/DefaultS3StorageConfigurationProvider.java:62]: misses `@NonNull` annotation

File: microstream/src/main/java/io/micronaut/microstream/s3/DefaultS3StorageConfigurationProvider.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        return bucketName;
+    }
+
+    /**
+     * Name of the bucket to use.
+     * @param bucketName
+     */
+    public void setBucketName(String bucketName) {
```
Comment (sdelamo) [inline microstream/src/main/java/io/micronaut/microstream/s3/DefaultS3StorageConfigurationProvider.java:70]: misses `@NonNull` annotation

File: microstream/src/test/resources/logging.properties [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Specify the handler, the handlers will be installed during VM startup.
```
Comment (sdelamo) [inline microstream/src/test/resources/logging.properties:1]: what is this file for? and why do we need it?

File: test-suite/build.gradle [hunk 5]
```diff
@@ -29,4 +30,15 @@ dependencies {
...
     testImplementation(projects.micronautMicrostreamRest)
     testImplementation(libs.jupiter.jupiter.params)
+
+    testImplementation(libs.managed.microstream.sql)
+    testImplementation(mnSql.micronaut.jdbc.hikari)
+    testRuntimeOnly(mnSql.postgresql)
+}
+
+micronaut {
+    testResources {
```
Comment (sdelamo) [inline test-suite/build.gradle:40]: I think we should disable platform bom with `importMicronautPlatform = false`. see: https://github.com/micronaut-projects/micronaut-sql/pull/891/files

**Review by sdelamo — COMMENTED** [review]
We miss test-suite/src/test/java/io/micronaut/microstream/docs/PostgresPersistentCacheTest.java and test-suite/src/test/java/io/micronaut/microstream/docs/PostgresCustomerControllerTest.java for S3

File: microstream/src/test/resources/logging.properties (hunk 4 above)
Comment (timyates) [reply microstream/src/test/resources/logging.properties:1]: Wasn't, it was a remnant of a failed attempt to debug postgres ;-)

File: test-suite/build.gradle (hunk 5 above)
Comment (timyates) [reply test-suite/build.gradle:40]: Done in https://github.com/micronaut-projects/micronaut-microstream/pull/190/commits/80db913866041bc103e75e1b98a470d9550a7443

**Comment by timyates** [comment]
@sdelamo I believe this is ready for review 🤞

File: src/main/docs/guide/storageTargets/s3.adoc [hunk 6]
```diff
@@ -0,0 +1,16 @@
+Configuration of an S3 storage target requires a https://micronaut-projects.github.io/micronaut-aws/latest/guide/#s3[S3Client bean to be defined].
```
Comment (sdelamo) [inline src/main/docs/guide/storageTargets/s3.adoc:1]:
```suggestion
Configuration of an https://docs.microstream.one/manual/storage/storage-targets/blob-stores/aws-s3.html[S3 storage target] requires a https://micronaut-projects.github.io/micronaut-aws/latest/guide/#s3[S3Client bean to be defined].
```

File: microstream/src/test/groovy/io/micronaut/microstream/s3/NamedS3StorageSpec.groovy [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        S3Config s3Config
+
+        LocalStackClient(S3Config s3Config) {
+            this.s3Config = s3Config
+        }
+
+        @Bean
+        @Singleton
+        @Named(OTHER_CLIENT_NAME)
+        S3Client buildClient() {
```
[the remark this answers is not in the document]
Comment (sdelamo) [reply microstream/src/test/groovy/io/micronaut/microstream/s3/NamedS3StorageSpec.groovy]: I have changed it here: https://github.com/micronaut-projects/micronaut-microstream/pull/190/commits/350e6d59af1b8a0eb3d309a2a1a1c0d8e098cfe4

File: test-suite-utils/src/main/java/io/microstream/testutils/LocalStackUtils.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
+package io.microstream.testutils;
```
Comment (timyates) [inline test-suite-utils/src/main/java/io/microstream/testutils/LocalStackUtils.java:1]: should this be `io.micronaut.microstream.testutils`?

Comment (timyates) [reply test-suite-utils/src/main/java/io/microstream/testutils/LocalStackUtils.java]: Renamed here https://github.com/micronaut-projects/micronaut-microstream/pull/190/commits/af2dcb7a55d9a5d6819e8c8587ce3ceaa65318b0

Comment (sdelamo) [reply test-suite-utils/src/main/java/io/microstream/testutils/LocalStackUtils.java]: thanks

**Pull request merged**